atlasdb-conjure/src/main/java/com/palantir/atlasdb/http/v2/ResultOrThrowable.java

```diff
@@ -30,8 +30,8 @@
 
     @Value.Check
     default void exactlyOneSet() {
-        Preconditions.checkState(result().isPresent() ^ throwable().isPresent()
-                || isSuccessful() && !throwable().isPresent());
+        Preconditions.checkState((result().isPresent() ^ throwable().isPresent())
+                || (isSuccessful() && throwable().isEmpty()));
     }
```

Comment:
this is a little hard to read - can we extract these 2 conditions out into methods?

Reply:
Sure, will do